src/spawn_stubs.c

```diff
@@ -488,6 +502,188 @@ CAMLprim value spawn_unix(value v_env,
     }
   }
 
+  if (new_mask || old_mask)
+    pthread_sigmask(SIG_SETMASK, new_mask, old_mask);
+}
+
+#if defined(USE_POSIX_SPAWN)
+
+CAMLprim value spawn_unix(value v_env,
+                          value v_cwd,
+                          value v_prog,
+                          value v_argv,
+                          value v_stdin,
+                          value v_stdout,
+                          value v_stderr,
+                          value v_use_vfork,
+                          value v_setpgid,
+                          value v_sigprocmask)
+{
+  CAMLparam4(v_env, v_cwd, v_prog, v_argv);
+  CAMLlocal1(e_arg);
+  e_arg = Nothing;
+
+  pid_t pid;
+  int tmp_fds[3] = {0};  // invariant: initialized > 2
+
+  int e_error;
+  char *e_function = NULL;
+
+  posix_spawn_file_actions_t actions;
+  if (posix_spawn_file_actions_init(&actions))
+    caml_raise_out_of_memory();
+
+  posix_spawnattr_t attr;
+  if (posix_spawnattr_init(&attr))
+    caml_raise_out_of_memory();
+
+  struct spawn_info info;
+  init_spawn_info(&info,
+                  v_env, v_cwd, v_prog, v_argv,
+                  v_stdin, v_stdout, v_stderr, v_setpgid);
+  init_spawn_info_sigmask(&info, v_sigprocmask, NULL, NULL);
+
+  short attr_flags = POSIX_SPAWN_SETSIGMASK;
+  if (info.set_pgid) attr_flags |= POSIX_SPAWN_SETPGROUP;
+  e_error = posix_spawnattr_setflags(&attr, attr_flags);
+  if (e_error) {
+    e_function = "setpgid";
```

Comment:
Shouldn't this say "posix_spawnattr_setflags"? (both because that's the function you're calling and because `POSIX_SPAWN_SETSIGMASK` is at least equally responsible, so we probably shouldn't blame setpgid.

Reply:
Yeah. I was doing something silly.